lib/src/components/NxThreatIndicatorLegend/NxThreatIndicatorLegend.tsx

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { forwardRef } from 'react';
+import classnames from 'classnames';
+import {
+  NxThreatIndicatorLegendProps,
+  nxThreatIndicatorLegendPropTypes,
+  nxThreatPropTypes,
+  ThreatProps
+} from './types';
+import { NxH3, NxThreatIndicator } from '../..';
+import './NxThreatIndicatorLegend.scss';
+
+const capitalize = (str: string) => str.charAt(0).toUpperCase() + str.slice(1);
+
+const ThreatIndicator = ({ threatType }: ThreatProps) => {
+  return (
+    <div>
+      <NxThreatIndicator threatLevelCategory={threatType} />
+      <span>{capitalize(threatType)}</span>
+    </div>
+  );
+};
+
+ThreatIndicator.propTypes = nxThreatPropTypes;
+
+const NxThreatIndicatorLegend = forwardRef<HTMLDivElement, NxThreatIndicatorLegendProps>(
+    (props: NxThreatIndicatorLegendProps, ref) => {
+      const {
+        critical,
+        severe,
+        moderate,
+        low,
+        none,
+        unspecified,
```

Comment:
I think we should consider the case where the user doesn't specify any of these props.  We might want to say that that's a more convenient way of specifying all of them, that when you don't set any, all are shown.  Or if not that, then I think we want to warn in the console and render nothing like what `NxThreatCounter` does when you don't specify any counts, or even be harsher and just throw an error.

Reply:
> I think we should consider the case where the user doesn't specify any of these props. We might want to say that that's a more convenient way of specifying all of them, that when you don't set any, all are shown. Or if not that, then I think we want to warn in the console and render nothing like what `NxThreatCounter` does when you don't specify any counts, or even be harsher and just throw an error.

Added `console.warn` (similar to `NxThreatCounter`'s) when no category level props are provided.
081ce53f7014d08a22dd07be1b317164a01301df